### PR TITLE
Add cases for -nbdkit checkwrite and blocksize_policy filters

### PR DESCRIPTION
**Pull request by mxie91**
Signed-off-by: Ming Xie <mxie@redhat.com>

### CODE REVIEW

**Comment by mxie91** [comment]
# avocado run --vt-type v2v --vt-machine-type q35 nbdkit.checkwrite
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 7a3c81be6cf251a6e9eca1b85fd2c4035a807447
JOB LOG    : /root/code/avocado/avocado/{{ base_dir }}/job-results/job-2022-11-23T22.34-7a3c81b/job.log
 (1/1) type_specific.io-github-autotest-libvirt.nbdkit.checkwrite: STARTED
 (1/1) type_specific.io-github-autotest-libvirt.nbdkit.checkwrite: PASS (3.13 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

JOB HTML   : /root/code/avocado/avocado/{{ base_dir }}/job-results/job-2022-11-23T22.34-7a3c81b/results.html
JOB TIME   : 53.91 s


# avocado run --vt-type v2v --vt-machine-type q35 nbdkit.blocksize_policy
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 1db65a9a17eae2762f373de86f5270ad961fd44d
JOB LOG    : /root/code/avocado/avocado/{{ base_dir }}/job-results/job-2022-11-24T00.24-1db65a9/job.log
 (1/1) type_specific.io-github-autotest-libvirt.nbdkit.blocksize_policy: STARTED
 (1/1) type_specific.io-github-autotest-libvirt.nbdkit.blocksize_policy: PASS (1.66 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/code/avocado/avocado/{{ base_dir }}/job-results/job-2022-11-24T00.24-1db65a9/results.html
JOB TIME   : 5.42 s

File: v2v/tests/src/nbdkit/nbdkit.py [hunk 1]
```diff
@@ -204,6 +205,36 @@ def pread(h, count, offset):
         if not re.search('config: error: Traceback', cmd_result.stderr_text):
             test.fail('failed to test enhance_python_error')
 
+    def test_checkwrite():
```
Comment (chunfuwen) [inline v2v/tests/src/nbdkit/nbdkit.py:208]: change name to test_checkwrite_filter() may be more exactly to describe its intension.

File: v2v/tests/src/nbdkit/nbdkit.py [hunk 2]
```diff
@@ -204,6 +205,36 @@ def pread(h, count, offset):
...
+        """
+        Mount nfs src to dst
+        """
+        nfs_src = params_get(params, "nfs_src")
+        mount_path = os.path.join(data_dir.get_tmp_dir())
+        utils_misc.mount(nfs_src, mount_path, 'nfs')
+        mount_file_path = os.path.join(mount_path, 'checkwrite_bound.sh')
+        if os.path.isdir(mount_file_path):
```
Comment (xiaodwan) [inline v2v/tests/src/nbdkit/nbdkit.py:216]: Don't use NFS server to manager the code. It's inconvenient to maintain.
The shell script could be generated during runtime.

File: v2v/tests/src/nbdkit/nbdkit.py (hunk 1 above)
Comment (mxie91) [reply v2v/tests/src/nbdkit/nbdkit.py]: updated, thanks!

File: v2v/tests/src/nbdkit/nbdkit.py (hunk 2 above)
Comment (mxie91) [reply v2v/tests/src/nbdkit/nbdkit.py]: Updated, pls check again, thanks!

File: v2v/tests/src/nbdkit/nbdkit.py [hunk 3]
```diff
@@ -204,6 +204,53 @@ def pread(h, count, offset):
...
+    echo 393216 655360 3
+  ;;
+  *) exit 2 ;;
+esac
+EOF
+
+sleep 1
+
+nbdsh -u nbd+unix:///?socket=/tmp/sock -c 'h.zero (655360, 262144, 0)'
+        """
```
Comment (xiaodwan) [inline v2v/tests/src/nbdkit/nbdkit.py:232]: Add a blank line to make it easier when reading.

Comment (mxie91) [reply v2v/tests/src/nbdkit/nbdkit.py:233]: Done, pls review again, thanks!

**Review by xiaodwan — APPROVED** [review]

**Pull request merged**